lmdeploy/pytorch/chat.py

```diff
@@ -1,158 +1,140 @@
-            use_fast_tokenizer should be True.
-        If using depodaca/llama-xb-hf, use_fast_tokenizer should be False.
-    """
+import itertools
+import logging
+from typing import Optional
 
-    if not _is_transformers_available:
-        raise ImportError('transformers is not installed.\n'
-                          'Please install with `pip install transformers`.\n')
-
-    tokenizer = AutoTokenizer.from_pretrained(tokenizer_path,
-                                              use_fast=use_fast_tokenizer,
-                                              trust_remote_code=True)
-
-    with LoadNoInit():
-        model = AutoModelForCausalLM.from_pretrained(model_path,
-                                                     torch_dtype=torch.float16,
-                                                     trust_remote_code=True)
-
-    if not _is_deepspeed_available:
-        warnings.warn('deepspeed is not installed, '
-                      'use plain huggingface model.')
-
-        model = model.cuda(local_rank)
-    else:
-        config = dict(
-            tensor_parallel=dict(tp_size=world_size),  # Number of GPU
-            dtype=torch.float16,  # dtype of the weights (fp16)
-            replace_with_kernel_inject=True,
-            # replace the model with the kernel injector
-            max_out_tokens=2048,
-        )
-        # For internlm model not supported by DeepSpeed,
-        # set replace_with_kernel_inject=False to use AutoTP.
-        # It's a hotfix before the progress is merged
-        # https://github.com/InternLM/lmdeploy/issues/136
-        if 'InternLM' in model.__class__.__name__:
-            try:
-                # Use customized deepspeed supporting internlm
-                # https://github.com/wangruohui/DeepSpeed/tree/support_internlm_0.10.0 (commit cdef2ce)  # noqa: E501
-                from deepspeed.module_inject.containers.internlm import \
-                    InternLMLayerPolicy
-            except ImportError:
-                # use stock deepspeed
-                config.update({'replace_with_kernel_inject': False})
-            else:
-                for module in model.modules():
-                    if module.__class__.__name__ == 'InternLMDecoderLayer':
-                        InternLMLayerPolicy._orig_layer_class = module.__class__  # noqa: E501
-                        break
-
-        model = deepspeed.init_inference(
-            model=model,  # Transformers models
-            config=config,
-        )
-
-    return tokenizer, model
+import fire
+import torch
+from transformers import GenerationConfig, PreTrainedModel
+
+from .adapters import init_adapter
+from .dist import get_local_rank, get_rank, get_world_size
+from .model import accel_model, init_model
+from .session import BasicSessionManager  # noqa: F401
+from .session import BasicSessionManagerWithHistory
+from .utils import BasicStreamer, TerminalIO, control
+
+logger = logging.getLogger(__name__)
```

Comment:
There is a logger module in lmdeploy/turbomind/utils.py. I think we can move it to 'lmdeploy/utils.py' so that both turbomind and pytorch can share it.

Reply:
Currently, logger in this module is for debug purpose. It will record a lot of tensor values. Since this is an interactive program in command line, such logs should only ne saved to files without printing to screen. However, mmlogger does not support only output to file, it only supports to screen or screen + file.